onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc

```diff
@@ -2283,6 +2321,34 @@ TEST(ReductionOpTest, ReduceDimWithZero) {
   test.AddInput<float>("data", {3, 0, 2}, {});
   test.AddOutput<float>("reduced", {1, 0, 1}, {});
   run(test);
+}
+
+TEST(ReductionOpTest, OptimizeShapeForFastReduce_ReduceDimWithZero2) {
+  FastReduceKind fast_kind;
+  std::vector<int64_t> fast_shape, fast_output_shape, fast_axes;
+  std::vector<int64_t> expected_fast_shape, expected_fast_output_shape, expected_fast_axes;
+
+  // R - keep_dims=1 - noop=false
+  fast_kind = OptimizeShapeForFastReduce(
```

Comment:
keep_dims is false in this case isn't it?

Reply:
done